LifeCycleAssessment_Engine/Compute/EvaluateEnvironmentalProductDeclaration.cs

```diff
@@ -0,0 +1,194 @@
+        [Input("BHoM Object", "This is a BHoM Object to calculate EPD off of. The method requires a volume property on the BHoM Object. Density is required if the chosen EPD is on a per mass basis, and will be extracted from the dataset if possible prior to extracting from the object itself.")]
+        [Input("environmentalProductDeclaration", "This is LifeCycleAssessment.EnvironmentalProductDeclaration data. Please select your desired dataset and supply your material choice to the corresponding BHoM objects.")]
+        [Input("environmentalProductDeclarationField", "Filter the provided EnvironmentalProductDeclaration by selecting one of the provided metrics for calculation. This method also accepts multiple fields simultaneously.")]
+        [Output("quantity", "The quantity of the desired metric provided by the EnvironmentalProductDeclarationField")]
+        public static double EvaluateEnvironmentalProductDeclarationPerObject(IBHoMObject obj, EnvironmentalProductDeclaration environmentalProductDeclaration, EnvironmentalProductDeclarationField environmentalProductDeclarationField = EnvironmentalProductDeclarationField.GlobalWarmingPotential) //default to globalWarmingPotential evaluation
+        {
+            if (obj != null)
+            {
+                //First check validity of inputs
+                object vol = obj.PropertyValue("Volume");
+                if (vol == null)
+                {
+                    BH.Engine.Reflection.Compute.RecordError("The input object must have a Volume property for this method to work.");
+                    return 0;
+                }
+                double volume = System.Convert.ToDouble(vol);
+                if (volume == 0)
+                {
+                    BH.Engine.Reflection.Compute.RecordError("The input object's Volume value is invalid. Volume should be in m3 in numerical format.");
+                    return 0;
+                }
+
+                //Check the epd declared unit type to determine whether to calculate by mass, volume, or neither
+                string declaredUnitType = environmentalProductDeclaration.DeclaredUnitType();
+
+                if (declaredUnitType == "Volume")
+                {
+                    return EvaluateEnvironmentalProductDeclarationByVolume(environmentalProductDeclaration, environmentalProductDeclarationField, volume);
+                }
+
+                else if (declaredUnitType == "Mass")
+                {
+                    double density = 0;
+                    string epdDensity = environmentalProductDeclaration.Density.ToString() ?? "";
+                    double densityVal = System.Convert.ToDouble(epdDensity.Substring(0, epdDensity.IndexOf(" ")));
```

Comment:
I appreciate there is a reason for doing this `substring` on the first space - but could we have a code comment that explains it for the next person that comes through to maintain this.